rfc/graphql-schema/additionalTypeDefs/byPage/shoppingLists.ts

```diff
@@ -0,0 +1,195 @@
+export default /* GraphQL */ `
+  type ShoppingListItemConnection {
+    collectionInfo: CollectionInfo
+  }
+
+  enum ShoppingListStatus {
+    DRAFT
+    ARCHIVED
+    READY_FOR_APPROVAL
+    APPROVED
+    REJECTED
+  }
+
+  # Assuming ShoppingList is not absorbed into an upgraded "Wishlist"
+  type ShoppingList implements Node {
+    id: ID!
+    name: String!
+    description: String
+    createdBy: Customer!
+    items: ShoppingListItemConnection!
+    updatedAt: DateTimeExtended!
+    # B2C shopping lists do not need a status
+    # we could split this into two entities OR
+    # think of B2C shopping lists as created in a (not shown) "approved" status
+    status: ShoppingListStatus!
+  }
+
+  type ShoppingListEdge {
+    node: ShoppingList!
+    cursor: String!
+  }
+
+  type ShoppingListConnection {
+    pageInfo: PageInfo!
+    collectionInfo: CollectionInfo
+    edges: [ShoppingListEdge!]!
+  }
+
+  extend type Customer {
+    shoppingLists(
+      before: String
+      after: String
+      first: Int
+      last: Int
+    ): ShoppingListConnection!
```

Comment:
I think this might also need the status & search filters.

Reply:
fixed:
- https://github.com/bigcommerce/b2b-buyer-portal/pull/401/commits/70ca532abfdfdd97f51b0084af42f3b9169ba24d